### DISCUSSION TITLE
codespace fails to build . builds locally

### DISCUSSION THREAD

**Discussion by solarslurpi** [community/community#50251] [Codespaces] [answered]
The devcontainer in [my repo](https://github.com/solarslurpi/mastoplaytime) fails to build. Yet, it build locally on my Windows 10 PC.  I find the log files to be overwhelming.  I am attempting to get the code space running on an Edge browser.  I have attached the log file.  What is the best way to debug this?  There is an overwhelming amount of text in the [log files where I uploaded to this location](https://github.com/solarslurpi/mastoplaytime/blob/bfeaa8dfad116325cfc6726ce0931917d6ee1d8d/myStuff/logs.zip).  Thank you.

**Comment by xec-abailey** [comment]
Yep getting the exact same error as of this morning. Was working last night perfectly fine but logged in today and can no longer build in a codespace: 

```
2023-03-16 16:59:02.889Z: Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-root/container-features/0.33.0-1678985938157/Dockerfile-with-features -t vsc-production-prediction-e4bfe8e86616502010a60d50ce5166d0624d4bc7662539eb3b21bbde6003ec45 --target dev_containers_target_stage --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.33.0-1678985938157 --build-arg _DEV_CONTAINERS_BASE_IMAGE=development --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /var/lib/docker/codespacemount/workspace/production-prediction
2023-03-16 16:59:02.893Z: Error Code: 1302
```

I checked through your creation.log and found the exact same error code related to docker-compose... something's fishy here

Quick edit: just to clarify this works fine when building to a local dev container like @solarslurpi stated as well

**Comment by xec-abailey** [comment]
After a bit more searching looks like something going on with the devcontainer.json features 


Had features disabled on my local devcontainer and it succeeded. Re-added them to the devcontainer like so: 

```
	"features": {
		"ghcr.io/devcontainers/features/docker-from-docker:1": {
			"version": "latest"
		}
	},
```

and received this error: 

```
ERROR: failed to solve: rpc error: code = Unknown desc = failed to solve with fr
ontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code
= Unknown desc = Syntax error - can't find = in "-a". Must be of the form: name=
value
```

**Reply by solarslurpi** [reply]
Thank you.  I was getting a bit overwhelmed.  I got so close... 😿

**Reply by xec-abailey** [reply]
Opened [this issue](https://github.com/devcontainers/features/issues/494) on devcontainer/features to see if we can get a true resolution. Unfortunately for me I need the features in my dev environment

**Comment by solarslurpi** [comment]
Thank you.

**Comment by jkeech** [comment, marked as answer]
@solarslurpi I think the issue is coming from your `docker-compose` config where it's referencing an `env_file` that doesn't exist in the repo. Can you try removing that line? I suspect that file exists locally which is why it builds fine on your machine, but since it's not checked into the repo, it fails to build on another machine.

https://github.com/solarslurpi/mastoplaytime/blob/bfeaa8dfad116325cfc6726ce0931917d6ee1d8d/.devcontainer/docker-compose.yml#L39

`docker-compose` is erroring with `Couldn't find env file: /var/lib/docker/codespacemount/workspace/mastoplaytime/config/.env`

**Reply by solarslurpi** [reply]
Thank you.

**Comment by xec-abailey** [comment]
For anyone that finds this later, super weird issue - posted what my ultimate fix was [here](https://github.com/devcontainers/features/issues/494#issuecomment-1472443328). Must have been a spec or parsing change with the devcontainer.json because I had an environmental variable that was causing the failure when the features were installed.
